### PR TITLE
Added a map coordinates overlay for the editor

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/7750.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+				Enabled ^= true;
+		}
+
+		public IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
+		{
+			if (!Enabled)
+				yield break;
+
+			var mouseCell = wr.Viewport.ViewToWorld(Viewport.LastMousePos).ToMPos(wr.World.Map).ToCPos(shape);
+			var cells = AllCells ? wr.Viewport.AllVisibleCells : new CellRegion(shape, mouseCell, mouseCell);
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs]: `AllCells` is never set, so is always false.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs]: I decided to tie this up to the chat command now.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs]: Okay, that is undocumented though.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs]: Not really:

> Optional parameter: 'allcells'"

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/World/MapCoordinatesOverlay.cs]: I had not seen that, sorry.

**Comment by pchote** [comment]
The useability of this could be improved a lot by combining it with the terrain grid visualization.

**Comment by pchote** [comment]
Elaborating on my earlier comment: a much simpler (both code an UX) implementation would be to show a text string "Cursor position: (x, y)" in a label widget at the top-left of the screen when the terrain grid is enabled.

**Comment by Phrohdoh** [comment]
I don't see why we can't show more than a single tile's coordinates, so not a fan of that requested change.

**Comment by pchote** [comment]
@Phrohdoh: Have you tested this ingame?  Our tile size is smaller than theme hospital, and our fonts are a lot chunkier, so the grid layout you posted in #7750 doesn't work in OpenRA.  To make them fit, @Mailaender displays the x and y coordinates as separate numbers (on separate rows), which hurts the usability of the feature.

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/issues/7750 wasn't about labels, but a grid overlay. I don't intend to change that.

**Comment by pchote** [comment]
The changes requested tag is to improve the poor usability of the feature:  The coordinates are not displayed in the standard x,y format, they are hidden by the mouse cursor, and become completely unreadable when you zoom out.  I have suggested a simple way to fix that, but I don't know why I bothered.

**Comment by Mailaender** [comment]
You told me in IRC afterwards that you did it yourself already combined in the terrain overlay so go ahead than. I don't have time for this UI bike shedding.

**Comment by Mailaender** [comment]
Superseded by https://github.com/OpenRA/OpenRA/pull/9089.

**Pull request closed without merging**